mesonbuild/backend/ninjabackend.py

```diff
@@ -326,6 +327,7 @@ def __init__(self, all_outputs: T.Set[str], outfilenames, rulename, infilenames,
         self.elems = []
         self.all_outputs = all_outputs
         self.output_errors = ''
+        self.use_rspfile = False
```

Comment:
I know in practice this doesn't race, but it creates a potential race condition, and that makes me nervous

Reply:
Are you really thinking of race conditions, or just potential logic bugs if `use_rspfile` is not initialized correctly? I thought of making this `T.Optional[bool]` for extra safety, but it looked just wrong.

Reply:
use_rspfile is initialized by calling one of it's methods, but read by another method. So if you called the methods out of order you'd get different results. so I guess more of a logic bug.

Reply:
An other option would be to implement caching inside the _should_use_rspfile method

Reply:
As the commit message says, the change relies on the fact that `count_rule_references()` will always be called before `write()`. Both are used only in one NinjaBackend method and this code doesn't use the threading or asyncio modules... there is no race condition.

But the requirement on that ordering could use directly being mentioned as a code comment since the codebase itself relies on this, not just the patch change rationale.